src/mds/Server.cc

```diff
@@ -1651,7 +1652,7 @@ void Server::reconnect_tick()
 		      << ", after waiting " << elapse1
 		      << " seconds during MDS startup";
 
-    if (g_conf()->mds_session_blacklist_on_timeout) {
+    if (g_conf()->mds_session_blacklist_on_timeout && !session->is_import_replayed()) {
```

Comment:
for this implementation, any session opened by replay_open_sessions will not get blacklisteded. It's not good.

Reply:
It will not get blacklisted only when replay_open_sessions is called in EImportStart journal replay: https://github.com/ceph/ceph/pull/36227/files#diff-e18d1bb1d5f6388430b7705cf32eac6dR2991, due to import_replayed boolean being set to true. In this scenario either 2 things happen: the session was not actually opened by the client (before the MDS failed) and so it gets killed (instead of blacklisted) during recconnect, or if the client had opened the session before the MDS went down then during reconnect the connection is established again and the boolean is set back to false: https://github.com/ceph/ceph/pull/36227/files#diff-eecf840e09d7481f7b8f415789852c7dR1501.

Is there an issue that you see with this approach?

Reply:
needs to handle the case that client has actually opened session before MDS went down and the client failed to reconnect

Reply:
@ukernel Tried working on this today but I'm still not sure how to handle the above case. Problem is the journalling for the opening of connection from the MDS side (ESession) comes after EImportStart is journalled. Now during failover journal replay, the MDS at EImportStart replay does not have any information on whether the session was really opened by the client or not. That information is in the next ESession replay. Any thoughts on how to handle this? I may be missing something here.

Reply:
for any session opened by EImportStart, set a client_not_open flag. If ESession is countered later, clear the flag.

Reply:
@ukernel Thanks! I missed your comment before. Clearing the flag on ESession seems like a good fix. I have updated the code.